fighters/common/src/function_hooks/camera.rs

```diff
@@ -9,8 +9,234 @@ unsafe fn normal_camera(ptr: u64, float: f32) {
     call_original!(ptr, float);
 }
 
+// Standardizes normal_camera_min_distance for all stages
+#[skyline::hook(offset = 0x26209bc, inline)]
+unsafe fn parse_stprm_normal_camera_min_distance(ctx: &mut skyline::hooks::InlineCtx) {
+    let normal_camera_min_distance: f32;
+    asm!("fmov w20, s0", out("w20") normal_camera_min_distance);
+    if normal_camera_min_distance < 120.0 {
+        let normal_camera_min_distance: f32 = 120.0;
+        asm!("fmov s0, w8", in("w8") normal_camera_min_distance);
+    }
+}
+
+// Standardizes swing_rate_x for all stages
+#[skyline::hook(offset = 0x2620c04, inline)]
+unsafe fn parse_stprm_swing_rate_x(ctx: &mut skyline::hooks::InlineCtx) {
+    let swing_rate_x: f32 = 0.0;
+    asm!("fmov s0, w8", in("w8") swing_rate_x)
+}
+
+// Standardizes swing_rate_y for all stages
+#[skyline::hook(offset = 0x2620cc8, inline)]
+unsafe fn parse_stprm_swing_rate_y(ctx: &mut skyline::hooks::InlineCtx) {
+    let swing_rate_y: f32 = 0.0;
+    asm!("fmov s0, w8", in("w8") swing_rate_y)
+}
+
+// Standardizes normal_camera_vertical_angle and normal_camera_look_down_vertical_angle for all stages
+#[skyline::hook(offset = 0x2620e54, inline)]
+unsafe fn parse_stprm_normal_camera_angles(ctx: &mut skyline::hooks::InlineCtx) {
+    let hash = *ctx.registers[18].x.as_ref();
+    let mut angle: f32 = 0.0;
+    asm!("fmov w20, s0", out("w20") angle);
+    if hash == hash40("normal_camera_vertical_angle") {
+        if angle < -5.0 {
+            angle = -5.0;
+        }
+    }
+    else if hash == hash40("normal_camera_look_down_vertical_angle") {
+        if angle < -25.0 {
+            angle = -25.0;
+        }
+    }
+    asm!("fmov s0, w8", in("w8") angle)
+}
+
+// Standardizes target_interpolation_rate for all stages
+#[skyline::hook(offset = 0x2620fec, inline)]
```

Comment:
There must be a better way to do this.

Are all of these hooks in the same function? Is it not possible to instead modify, at the end of the function, the location in memory they are stored?